### ISSUE TITLE
Viewing test coverage over time

### ISSUE THREAD

**Issue by borrascador** [istanbuljs/nyc#1141]
Is there a way to use this tool to generate a time-series of reports based on the git history of a project? I'm picturing generating a coverage report for each commit in a given range and the output would be the same as the current html output but with a slider added to view coverage at different points in time.

Not sure if this is a feature request or not, because hopefully somebody's tried to implement it already.

**Comment by bcoe** [comment]
@borrascador you could potentially build this by checking out your repo at a variety of points, and executing tests with coverage -- one nice thing about nyc is that it's relatively configuration free.

You would simply need to then set aside the `coverage` output from each run and then plot it.

tldr; not a feature we're going to add to `nyc`, but I think you should be able to pull it off pretty easily.

**Comment by bcoe** [comment]
I'd potentially use the `json` report, or something similar, which will give you raw data you can map.

**Comment by nadavye** [comment]
Look at http://www.sealights.io
See if that helps you.

**Comment by Ryuno-Ki** [comment]
Look at [Gource](https://github.com/acaudwell/Gource)
[Demo with Linux Kernel history](https://www.youtube.com/watch?v=P_02QGsHzEQ)

**Comment by borrascador** [comment]
Thanks for the suggestions. I think I will go with the suggestion by @bcoe to export `json` for each commit and then map the data into something useful.
